File: apps/models/Wilayah.py
from apps.conn2 import db
from apps.models import BaseModel
class Wilayah1(BaseModel):
    __tablename__ = 'wilayah1'
    # id = db.Column(db.Integer, primary_key=True)
    # nama = db.Column(db.String(100))

    nama = BaseModel.string(100)

class Wilayah2(BaseModel):
    __tablename__ = 'wilayah2'
    # id = db.Column(db.Integer, primary_key=True)
    # nama = db.Column(db.String(100))
    # id_wilayah1 = db.Column(db.Integer, db.ForeignKey('wilayah1.id'))

    nama = BaseModel.string(100)
    id_wilayah1 = BaseModel.foreign('wilayah1.id')

class Wilayah3(BaseModel):
    __tablename__ = 'wilayah3'
    # id = db.Column(db.Integer, primary_key=True)
    # nama = db.Column(db.String(100))
    # id_wilayah2 = db.Column(db.Integer, db.ForeignKey('wilayah2.id'))

    nama = BaseModel.string(100)
    id_wilayah2 = BaseModel.foreign('wilayah2.id')

class Wilayah4(BaseModel):
    __tablename__ = 'wilayah4'
    # nama = db.Column(db.String(100))
    # id_wilayah3 = db.Column(db.Integer, db.ForeignKey('wilayah3.id'))

    id = db.Column(db.BigInteger, primary_key=True)
    nama = BaseModel.string(100)
    id_wilayah3 = BaseModel.foreign('wilayah3.id')

File: 8budimas-purchase/app/Models/Wilayah.php
<?php

namespace App\Models;
use Illuminate\Support\Facades\Session;

class Wilayah extends Model
{
    /**
     * Mendapatkan Semua Data Wilayah1.
     * @return object (N) Data Wilayah1.
     */
    function getProvinsi() {
        return $this->orderBy('id')->select('/api/base/wilayah1')->get();
    }

    /**
     * Mendapatkan Data Wilayah2 Berdasarkan Id Wilayah1.
     * 
     * @param integer|$id id_wilayah1.
     * @return object (N) Data Wilayah2.
     */ 
    function getKabupatenKota($id) {
        return $this->orderBy('id')->where(["id_wilayah1", "=", $id])
               ->select('/api/base/wilayah2')->get();
    }

    /**
     * Mendapatkan Data Wilayah3 Berdasarkan Id Wilayah2.
     * 
     * @param integer|$id id_wilayah2.
     * @return object (N) Data Wilayah3.
     */  
    function getKecamatan($id) {
        return $this->orderBy('id')->where(["id_wilayah2", "=", $id])
               ->select('/api/base/wilayah3')->get();
    }
    
    /**
     * Mendapatkan Data Wilayah4 Berdasarkan Id Wilayah3.
     * 
     * @param integer|$id id_wilayah3.
     * @return object (N) Data Wilayah4.
     */  
    function getKelurahan($id) {
        return $this->orderBy('id')->where(["id_wilayah3", "=", $id])
               ->select('/api/base/wilayah4')->get();
    }
}